test/wallet-test.js

```diff
@@ -2051,6 +2065,190 @@ describe('Wallet', function() {
+      };
+      await wallet.txdb.add(register, block);
+
+      // Check
+      bal = await wallet.getBalance();
+      assert.strictEqual(bal.tx, 5);
+      assert.strictEqual(bal.coin, 4);
+      assert.strictEqual(bal.confirmed, 10e6 - (4 * fee));
+      assert.strictEqual(bal.unconfirmed, 10e6 - (4 * fee));
+      assert.strictEqual(bal.ulocked, secondHighest);
+      assert.strictEqual(bal.clocked, secondHighest);
+    });
+
+    it('should send and confirm TRANSFER', async () => {
+      const recipAddr = await recip.receiveAddress();
+      const transfer = await wallet.sendTransfer(name, recipAddr, {hardFee: fee});
+
+      // Check
+      let senderBal = await wallet.getBalance();
+      assert.strictEqual(senderBal.tx, 6);
+      assert.strictEqual(senderBal.coin, 4);
+      assert.strictEqual(senderBal.confirmed, 10e6 - (4 * fee));
+      assert.strictEqual(senderBal.unconfirmed, 10e6 - (5 * fee));
+      assert.strictEqual(senderBal.ulocked, secondHighest);
+      assert.strictEqual(senderBal.clocked, secondHighest);
+
+      let recipBal = await recip.getBalance();
+      assert.strictEqual(recipBal.tx, 0);
+      assert.strictEqual(recipBal.coin, 0);
+      assert.strictEqual(recipBal.confirmed, 0);
+      assert.strictEqual(recipBal.unconfirmed, 0);
+      assert.strictEqual(recipBal.ulocked, 0);
+      assert.strictEqual(recipBal.clocked, 0);
+
+      // Confirm TRANSFER
+      const block = {
+        height: wdb.height + 1,
+        hash: Buffer.alloc(32),
+        time: Date.now()
+      };
+      await wallet.txdb.add(transfer, block);
+      await recip.txdb.add(transfer, block);
+
+      // Check
+      senderBal = await wallet.getBalance();
+      assert.strictEqual(senderBal.tx, 6);
+      assert.strictEqual(senderBal.coin, 4);
+      assert.strictEqual(senderBal.confirmed, 10e6 - (5 * fee));
+      assert.strictEqual(senderBal.unconfirmed, 10e6 - (5 * fee));
+      assert.strictEqual(senderBal.ulocked, secondHighest);
+      assert.strictEqual(senderBal.clocked, secondHighest);
+
+      recipBal = await recip.getBalance();
+      assert.strictEqual(recipBal.tx, 0);
+      assert.strictEqual(recipBal.coin, 0);
+      assert.strictEqual(recipBal.confirmed, 0);
+      assert.strictEqual(recipBal.unconfirmed, 0);
+      assert.strictEqual(recipBal.ulocked, 0);
+      assert.strictEqual(recipBal.clocked, 0);
+    });
+
+    it('should send and confirm FINALIZE', async () => {
+      // Advance past lockup period
+      wdb.height += network.names.transferLockup;
+
+      const finalize = await wallet.sendFinalize(name, {hardFee: fee});
+
+      // Check
+      let senderBal = await wallet.getBalance();
+      assert.strictEqual(senderBal.tx, 7);
+      // One less wallet coin because name UTXO belongs to recip now
+      assert.strictEqual(senderBal.coin, 3);
+      assert.strictEqual(senderBal.confirmed, 10e6 - (5 * fee));
+      assert.strictEqual(senderBal.unconfirmed, 10e6 - secondHighest - (6 * fee));
```

Comment:
I like the way that you use `hardFee`, maybe this will be less brittle if you use a global variable instead of hardcoding in the number of txs to multiply against the fee, then you can do something like `(txCount++ * fee)`